### DISCUSSION TITLE
Error 0x80070491 in MddBootstrapInitialize(0x00010000, experimental1, 0.0.0.0)

### DISCUSSION THREAD

**Discussion by yangxiaopo** [microsoft/WindowsAppSDK#1325] [Q&A] [answered]
With this guide, [https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/tutorial-unpackaged-deployment?tabs=cpp](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/tutorial-unpackaged-deployment?tabs=cpp), I have run successfully but **Error 0x80070491 in MddBootstrapInitialize(0x00010000, experimental1, 0.0.0.0)**.
![image](https://user-images.githubusercontent.com/16287565/131061907-da3e51c3-4cbb-4c07-9457-18b9064d52bb.png)
![image](https://user-images.githubusercontent.com/16287565/131061981-36c9bda6-624b-42cb-906f-770ed9b52009.png)

**Comment by riverar** [comment, marked as answer]
Did you also install the Windows App Runtime? Find and execute the appropriate Microsoft.WindowsAppSDK.Installer-[...] in this zip file: https://download.microsoft.com/download/b/8/b/b8b4508e-ea29-4381-85d9-69b824151f8d/Microsoft.WindowsAppSDK.Redist.1.0.0-exp.zip

**Reply by yangxiaopo** [reply]
I have installed ProjectReunion 0.8 stable and then install 1.0 Experimental version of the Windows App SDK for building and deploying an unpackaged app. this could be runtime mismatched after checking for installed versions of the Windows App SDK runtime.
![image](https://user-images.githubusercontent.com/16287565/131064341-00a9e750-3ef3-412f-a9d6-43a8d5e533d7.png)

**Reply by DrusTheAxe** [reply]
> MddBootstrapInitialize(0x00010000, experimental1, 0.0.0.0)

That will look for any 1.0-experimental1 packages (technically where the MSIX version is >= 0.0.0.0, i.e. any). Do you have 1.0-experimental1 registered? You can tell with

```
powershell -c get-appxpackage *appsdk*
```

That should find...

* Microsoft.WindowsAppSDK.1.0-experimental1
* Microsoft.WindowsAppSDK.Main.1.0-experimental1
* Microsoft.WindowsAppSDK.DDLM.1.0* (I don't recall exact package Name, it includes the MSIX version, plus architecture and short version tag, e.g. "e1" for "experimental1")

If not then the Bootstrap API won't succeed.

0.8 won't matter here. Your `MddBootstrapInitialize` call won't match 0.8 packages. They have different package Names so it doesn't matter if both 0.8 and 1.0 have an 'M' or a '0' in their name, either they're identical or they're not.

**Reply by btueffers** [reply]
@yangxiaopo if this resolved your issue please mark it as the answer!
